pkg/cmd/browse/browse.go

```diff
@@ -113,7 +113,9 @@ func NewCmdBrowse(f *cmdutil.Factory, runF func(*BrowseOptions) error) *cobra.Co
 			}
 
 			if err := cmdutil.MutuallyExclusive(
-				"specify only one of `--branch`, `--commit`, `--projects`, `--releases`, `--settings`, or `--wiki`",
+				"specify only one of `number arg`, `commit SHA arg`, `--branch`, `--commit`, `--projects`, `--releases`, `--settings`, or `--wiki`",
+				isNumber(opts.SelectorArg),
+				isCommit(opts.SelectorArg),
```

Comment:
I feel like this error message and error condition is getting quiet long and convoluted. I would rather see this new error condition broken out into its own check with a nicer error message for the user. Something like:

```go
      if (isNumber(opts.SelectorArg) || isCommit(opts.SelectorArg)) && (opts.Branch != "" || opts.Commit != "") {
        return cmdutil.FlagErrorf("%q is an invalid argument when using `--branch` or `--commit`", opts.SelectorArg)
      }
```